app/javascript/components/record-form/form/subforms/subform-fields/components/violation-item/component.jsx

```diff
@@ -26,7 +108,57 @@ const Component = ({ fields, values, locale, displayName, index, collapsedFieldV
         </div>
       }
     >
+      {canVerify && mode.isShow ? (
+        <Button
+          onClick={event => handleOpenVerifyModal(index, event)}
```

Comment:
Props that are objects, calculated, functions should be in a var and referenced as the value of the prop. Leaving an object in a prop causes unnecessary re-renders. Need to change in a number of places

Reply:
i18n.t is fine